From 63a1fe2560d30c2e1493fa9d7a76e2d980bf3473 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:04 -0400
Subject: [usb] Make DELAY_INIT quirk wait 100ms between Get Configuration requests

Message-id: <1410887463-20674-353-git-send-email-dzickus@redhat.com>
Patchwork-id: 95367
O-Subject: [RHEL7 PATCH 352/591] usb: Make DELAY_INIT quirk wait 100ms between Get Configuration requests
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d86db25e53fa69e3e97f3b55dd82a70689787c5d
Author: Julius Werner <jwerner@chromium.org>
Date:   Tue Mar 4 11:27:38 2014 -0800

    usb: Make DELAY_INIT quirk wait 100ms between Get Configuration requests

    The DELAY_INIT quirk only reduces the frequency of enumeration failures
    with the Logitech HD Pro C920 and C930e webcams, but does not quite
    eliminate them. We have found that adding a delay of 100ms between the
    first and second Get Configuration request makes the device enumerate
    perfectly reliable even after several weeks of extensive testing. The
    reasons for that are anyone's guess, but since the DELAY_INIT quirk
    already delays enumeration by a whole second, wating for another 10th of
    that isn't really a big deal for the one other device that uses it, and
    it will resolve the problems with these webcams.

    Signed-off-by: Julius Werner <jwerner@chromium.org>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/config.c b/drivers/usb/core/config.c
index 14ba398..1ab4df1 100644
--- a/drivers/usb/core/config.c
+++ b/drivers/usb/core/config.c
@@ -716,6 +716,10 @@ int usb_get_configuration(struct usb_device *dev)
    result = -ENOMEM;
    goto err;
   }
+
+  if (dev->quirks & USB_QUIRK_DELAY_INIT)
+   msleep(100);
+
   result = usb_get_descriptor(dev, USB_DT_CONFIG, cfgno,
       bigbuffer, length);
   if (result < 0) {
-- 
1.7.1